spec/lib/api/v3/work_packages/work_package_representer_spec.rb

```diff
@@ -1151,6 +1207,39 @@
         end
       end
 
+      describe 'fileLinks' do
+        let(:storages_feature_enabled) { true }
+        let(:storage) { build_stubbed(:storage) }
+        let(:file_link) { build_stubbed(:file_link, storage: storage, container: work_package) }
+
+        before do
+          allow(OpenProject::FeatureDecisions).to receive(:storages_module_active?).and_return(storages_feature_enabled)
+          allow(work_package).to receive(:file_links).and_return([file_link])
+        end
+
+        it 'embeds a collection' do
+          is_expected
+            .to be_json_eql('Collection'.to_json)
+                  .at_path('_embedded/fileLinks/_type')
+        end
+
+        it 'embeds with an href containing the work_package' do
+          is_expected
+            .to be_json_eql(api_v3_paths.file_links(work_package.id).to_json)
+                  .at_path('_embedded/fileLinks/_links/self/href')
+        end
+
+        it 'embeds the visible file links' do
+          is_expected
+            .to be_json_eql(1.to_json)
+                  .at_path('_embedded/fileLinks/total')
+
+          is_expected
+            .to be_json_eql(api_v3_paths.file_link(file_link.id).to_json)
+                  .at_path('_embedded/fileLinks/_embedded/elements/0/_links/self/href')
+        end
+      end
```

Comment:
The cases where this is not displayed are not speced.

Reply:
the cases where the embedded resources are not isplayed are usually tested from the links section. The shared example of `has no link` has a sub test `has no embedded resource`. Does this suffice?